control-plane/cni/main_test.go

```diff
@@ -55,40 +58,375 @@ func TestRun_cmdAdd(t *testing.T) {
 			expectedErr: fmt.Errorf("must be called as final chained plugin"),
 		},
 		{
-			name:      "Missing IPs in prevResult in stdin data, should throw error",
+			name: "Missing IPs in prevResult in stdin data, should throw error",
+			cmd: &Command{
+				client: fake.NewSimpleClientset(),
+			},
 			podName:   "corrupt-prev-result",
 			stdInData: missingIPsStdinData,
-			configuredPod: func(pod *corev1.Pod) *corev1.Pod {
+			configuredPod: func(pod *corev1.Pod, cmd *Command) *corev1.Pod {
 				_, err := cmd.client.CoreV1().Pods(defaultNamespace).Create(context.TODO(), pod, metav1.CreateOptions{})
 				require.NoError(t, err)
 
 				return pod
 			},
 			expectedErr: fmt.Errorf("got no container IPs"),
 		},
+		{
+			name: "Pod with incorrect traffic redirection annotation, should throw error",
+			cmd: &Command{
+				client: fake.NewSimpleClientset(),
+			},
+			podName:   "pod-with-incorrect-annotation",
+			stdInData: goodStdinData,
+			configuredPod: func(pod *corev1.Pod, cmd *Command) *corev1.Pod {
```

Comment:
It could go either way really. What I like about this way is that everything is all in one place for the test case for what I am doing with the pod. Like look at this scenario where I have to add the iptables config:

```
				cfg := iptables.Config{
					ProxyUserID:      "123",
					ProxyInboundPort: 20000,
					ConsulDNSIP:      "10.0.34.16",
				}
				iptablesConfigJson, err := json.Marshal(&cfg)
				require.NoError(t, err)
				pod.Annotations[annotationRedirectTraffic] = string(iptablesConfigJson)

```

I could put that iptables.Config in it's own test case field and update the pod before running the command.

Tell me what you think is clearer and I will make it happen. 😄

Reply:
Ah got it! thanks for clarifying!